operator/controllers/keptnworkloadinstance/controller.go

```diff
@@ -128,8 +128,13 @@ func (r *KeptnWorkloadInstanceReconciler) Reconcile(ctx context.Context, req ctr
 	appPreEvalStatus := appVersion.Status.PreDeploymentEvaluationStatus
 	if !appPreEvalStatus.IsSucceeded() {
 		if appPreEvalStatus.IsFailed() {
+			//cancell everything, as app pre-eval tasks have failed
+			if err := r.cancelWorkloadInstanceActions(ctx, workloadInstance, phase); err != nil {
+				span.SetStatus(codes.Error, err.Error())
+				return ctrl.Result{Requeue: true}, err
+			}
 			controllercommon.RecordEvent(r.Recorder, phase, "Warning", workloadInstance, "Failed", "has failed since app has failed", workloadInstance.GetVersion())
-			return ctrl.Result{Requeue: true, RequeueAfter: 20 * time.Second}, nil
```

Comment:
this has to be requeued, as otherwise this wouldn't be restarted when the pre-deployment tasks/evaluations of the app are corrected. It would make sense to go for an exponential backoff here